open-api/rest-catalog-open-api.yaml

```diff
@@ -410,7 +410,17 @@ paths:
         204:
           description: Success, no content
         400:
-          $ref: '#/components/responses/BadRequestErrorResponse'
+          description:
+            A bad request error or a NamespaceNotEmptyException in case the namespace is not empty
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IcebergErrorResponse'
+              examples:
+                BadRequestExample:
+                  $ref: '#/components/examples/BadRequestError'
+                NamespaceNotEmptyExample:
```

Comment:
Why use 400 for this and not a better error code?

Reply:
The current server behavior would be to return a 400 and the goal here was to document that behavior. Going forward we could add a new error code to indicate the `NamespaceNotEmptyException` but clients would still have to look at the error type of the 400 when talking to legacy REST servers

Reply:
@rdblue I've updated this to throw a 409 when a namespace isn't empty